### PR TITLE
feat: display test results in build scan detail view

### LINKED ISSUE #57: Display test results in build scan detail view
## Summary

Add a test results section to the build scan detail page, following the same pattern used for tasks. The GraphQL schema already exposes `tests`, `testCount`, and the `TestCaseConnection` with Relay pagination (added in PR #52).

## Current state

- Backend: fully implemented — `Test` type with `className`, `methodName`, `executorName`, `outcome` fields, exposed via GraphQL with cursor pagination
- Frontend: no test display — `scan-detail.component.ts` only shows the tasks table

### GraphQL query changes

Extend the `GET_BUILD_SCAN` query in `scan-detail.component.ts` to include:
```graphql
testCount
tests(first: $firstTests, after: $afterTests) {
  edges {
    cursor
    node {
      id
      className
      methodName
      executorName
      outcome
    }
  }
  pageInfo {
    hasNextPage
    endCursor
  }
  totalCount
}
```

### UI

Add a "Tests" section below the existing "Tasks" table in the scan detail view:
- Table columns: Class Name, Method Name, Outcome, Executor
- Outcome badges following the same pattern as tasks (green=Passed, red=Failed, yellow=Skipped)
- Paginated with "Load More" (same pattern as tasks)
- Show `testCount` in the scan summary area alongside `taskCount`
- Show "No tests" message when `testCount` is 0

### Scan list changes

Optionally add `testCount` to the scan list query and display it alongside `taskCount` in `scan-list.component.ts`.

## Related

- PR #52 — added the backend test event decoding and GraphQL schema
- Issue #53 — test outcome not yet populated (will show as null until fixed)
- Issue #54 — test executorName not yet populated

### PR DESCRIPTION
**Pull request by tacascer**
## Summary

- Add a **Tests** table to the scan detail page with columns: Class Name, Method Name, Outcome, Executor
- Outcome badges follow the same pattern as tasks: green (Passed), red (Failed), yellow (Skipped)
- Tests section is hidden when `testCount` is 0
- Add `testCount` column to the scan list view alongside `taskCount`
- Use `taskCount`/`testCount` fields in scan detail header instead of `tasks.totalCount`
- Split pagination variables (`$firstTasks/$afterTasks`, `$firstTests/$afterTests`) to support independent pagination later

## Test plan

- [x] 11 new tests for `scan-detail.component` covering test table rendering, badge colors, null field handling
- [x] 4 new tests for `scan-list.component` covering testCount column display
- [x] All 16 tests passing via `bazel test //angular/...`

Closes #57

### CODE REVIEW

File: angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts [hunk 1]
```diff
@@ -40,6 +42,23 @@ const GET_BUILD_SCAN = gql`
         }
         totalCount
```
[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts]: Good catch — removed `totalCount` from the `tasks` connection since `taskCount` is now used directly.
